### PR TITLE
Fix SSIMLoss when batchsize bigger than 2

### LINKED ISSUE #5886: SSIM loss fail when batchsize>1
**Describe the bug**
SSIM loss fail when batchsize>1

**To Reproduce**
```
import torch
from monai.losses.ssim_loss import SSIMLoss
x = torch.ones([4,1,10,10])/2
y = torch.ones([4,1,10,10])/2
data_range = x.max().unsqueeze(0)
print(1-SSIMLoss(spatial_dims=2)(x,y,data_range))
```

**Expected behavior**
Should print 1, but got error

**Screenshots**

**Environment**

Ensuring you use the relevant python executable, please paste the output of:

```
python -c 'import monai; monai.config.print_debug_info()'
```

**Additional context**

### PR DESCRIPTION
**Pull request by jak0bw**
Signed-off-by: Jakob Weigand <jakob.weigand@tum.de>

Fixes #5886.

### Description

When SSIMLoss is called with a tensor of batchsize > 1, SSIMMetric is called for each batch item individually. The result of each call to SSIMMetric is stored in ssim_val. Subsequently, ssim_val is concatenated to ssim_value, which is a collection of all calls to SSIMMetric. Before the concatenation both ssim_val and ssim_value are given a new view of size 1. As ssim_value collects the results of all earlier calls to SSIMMetric, .view(1) raises an error for the second call (as ssim_value has a different absolute size). This commit changes the behavior of the view call, to always use the current size of ssim_value, avoiding this issue but keeping the behavior to concatenate in a single row vector.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — APPROVED** [review]
Thank you, it looks good to me. Im merging this PR, @Can-Zhao please also help verify and feel free to reopen the discussion if needed.
